File: app/src/main/kotlin/tech/mmarca/openvitals/domain/query/SleepPeriodData.kt
package tech.mmarca.openvitals.domain.query

import tech.mmarca.openvitals.domain.model.DailySleepDuration
import tech.mmarca.openvitals.domain.model.SleepData

data class SleepPeriodData(
    val sessions: List<SleepData> = emptyList(),
    val previousSessions: List<SleepData> = emptyList(),
    val baselineSessions: List<SleepData> = emptyList(),
    val dailyDurations: List<DailySleepDuration> = emptyList(),
    val previousDailyDurations: List<DailySleepDuration> = emptyList(),
    val baselineDailyDurations: List<DailySleepDuration> = emptyList(),
)

File: lib/domain/query/sleep_period_data.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../model/sleep_models.dart';

part 'sleep_period_data.freezed.dart';

@freezed
abstract class SleepPeriodData with _$SleepPeriodData {
  const factory SleepPeriodData({
    @Default(<SleepData>[]) List<SleepData> sessions,
    @Default(<SleepData>[]) List<SleepData> previousSessions,
    @Default(<SleepData>[]) List<SleepData> baselineSessions,
    @Default(<DailySleepDuration>[]) List<DailySleepDuration> dailyDurations,
    @Default(<DailySleepDuration>[])
    List<DailySleepDuration> previousDailyDurations,
    @Default(<DailySleepDuration>[])
    List<DailySleepDuration> baselineDailyDurations,
  }) = _SleepPeriodData;
}
